client/src/templates/Introduction/components/block.test.tsx

```diff
@@ -0,0 +1,182 @@
+import * as getEnvData from '../../../utils/get-envdata';
+import Block from './block';
+const defaultProps = {
+  blockDashedName: 'test-block',
+  challenges: [
+    {
+      challenge: {
+        block: 'testblock',
+        certification: 'mockCertification',
+        challengeOrder: 1,
+        challengeType: 2,
+        dashedName: 'mock-dashed-name',
+        description: 'mockDescription',
+        challengeFiles: {} as ChallengeFiles,
+        fields: {} as Fields,
+        forumTopicId: 12345,
+        guideUrl: 'https://mockurl.com',
+        head: ['mockHead'],
+        hasEditableBoundaries: false,
+        helpCategory: 'mockHelpCategory',
+        id: 'mockId',
+        instructions: 'mockInstructions',
+        isComingSoon: false,
+        internal: {
+          content: 'mockContent',
+          contentDigest: 'mockContentDigest',
+          description: 'mockInternalDescription',
+          fieldOwners: ['mockOwner'],
+          ignoreType: null,
+          mediaType: 'mockMediaType',
+          owner: 'mockOwner',
+          type: 'mockType'
+        },
+        notes: 'mockNotes',
+        prerequisites: [] as PrerequisiteChallenge[],
+        removeComments: false,
+        isLocked: false,
+        isPrivate: false,
+        order: 1,
+        question: {} as Question,
+        assignments: ['mockAssignment'],
+        required: [],
+        solutions: {
+          ['indexhtml']: {} as FileKeyChallenge,
+          ['scriptjs']: {} as FileKeyChallenge,
+          ['stylescss']: {} as FileKeyChallenge,
+          ['indexjsx']: {} as FileKeyChallenge
+        },
+        sourceInstanceName: 'mockSourceInstanceName',
+        superOrder: 1,
+        superBlock: SuperBlocks.UpcomingPython,
+        tail: ['mockTail'],
+        template: 'mockTemplate',
+        tests: [] as Test[],
+        time: 'mockTime',
+        title: 'mockTitle',
+        translationPending: false,
+        url: 'https://mockurl.com',
+        usesMultifileEditor: false,
+        videoId: 'mockVideoId',
+        videoLocaleIds: {},
+        bilibiliIds: {} as BilibiliIds,
+        videoUrl: 'https://mockvideourl.com'
+      }
+    }
+  ],
+  completedChallengeIds: ['testchallengeIds'],
+  isExpanded: false,
+  t: jest.fn((key: string) => [key]),
+  toggleBlock: jest.fn()
+};
+
+const mockedEnvData = {
+  curriculumLocale: 'english',
+  showUpcomingChanges: false,
+  showNewCurriculum: false,
+  homeLocation: '',
+  apiLocation: '',
+  forumLocation: '',
+  newsLocation: '',
+  radioLocation: '',
+  clientLocale: 'english',
+  showLocaleDropdownMenu: false,
+  deploymentEnv: '',
+  environment: '',
+  algoliaAppId: '',
+  algoliaAPIKey: '',
+  stripePublicKey: null,
+  paypalClientId: null,
+  patreonClientId: null,
+  sentryClientDSN: null,
+  growthbookUri: null
+};
+
+describe('<Block />', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  const langaugesArr = Object.values(Languages);
```

Comment:
All languages are more or less equivalent, except that they differ in the value of `isAudited`. Rather than testing every single superblock of every language, it's sufficient to test a single language, superblock pair.

Then you can mock isAudited and test what happens when it returns true and when it returns false.

Reply:
Based on previous discussions, I had implemented the changes suggested. However, I understand your additional concerns and will make the necessary updates accordingly. Thank you for your guidance.

Reply:
Thanks @Riya267, I appreciate that. Sometimes it takes a few iterations to land on a nice solution, so thanks for bearing with us while we go through the process.

Reply:
Hey @ojeytonwilliams  I have updated the code based on our discussion. kindly review. 
Thanks